Mage/src/main/java/mage/abilities/common/EntersBattlefieldAllTriggeredAbility.java

```diff
@@ -85,20 +78,17 @@ public boolean checkEventType(GameEvent event, Game game) {
     public boolean checkTrigger(GameEvent event, Game game) {
         UUID targetId = event.getTargetId();
         Permanent permanent = game.getPermanent(targetId);
-        if (!filter.match(permanent, getControllerId(), this, game)) {
-            return false;
-        }
         this.getEffects().setValue("permanentEnteringBattlefield", permanent);
         this.getEffects().setValue("permanentEnteringControllerId", permanent.getControllerId());
         switch (setTargetPointer) {
             case PLAYER:
                 this.getEffects().setTargetPointer(new FixedTarget(permanent.getControllerId()));
-                return true;
+                break;
             case PERMANENT:
                 this.getEffects().setTargetPointer(new FixedTarget(permanent, game));
             default:
-                return true;
         }
+        return filter.match(permanent, getControllerId(), this, game);
```

Comment:
I'm concerned by this change, because now effect values are getting set even when it doesn't end up triggering. I don't actually know for sure if this causes an issue, but no other trigger does that and so I'd prefer to avoid it.

Can you think of a way to fix it, maybe using that thisOrAnother boolean, so that the values are only ever set when the trigger returns true?

Reply:
I will have a think. One idea is to separate the setValue & setTargetPointer parts into their own function which has to be called (i.e. you have to remember to do it) by classes that extend `EntersBattlefieldAllTriggeredAbility` and override `checkTrigger`. It's kind of like that now in that you have to remember to call `super.checkTrigger`.

I'm going to sleep on this and we'll see what we can come up with.

Reply:
I think I like the way you've done it now. Going to contemplate it a little more before approving.